mgplvm/manifolds/torus.py

```diff
@@ -100,31 +100,20 @@ def gmul(x: Tensor, y: Tensor) -> Tensor:
         return x + y
 
     @staticmethod
-    def distance(x: Tensor, y: Tensor) -> Tensor:
+    def distance(x: Tensor, y: Tensor, ell: Tensor = torch.ones(1, 1,
+                                                                1)) -> Tensor:
         # distance = 2 - 2 cos(x-y)
         # here we use the identity: cox(x-y) = cos(x)cos(y) + sin(x)sin(y)
         d = x.shape[-2]
-        cx = torch.cos(x)
-        cy = torch.cos(y)
-        sx = torch.sin(x)
-        sy = torch.sin(y)
-        z1_ = torch.cat([cx, sx], dim=-2)
-        z2_ = torch.cat([cy, sy], dim=-2)
-        res = 2 * (d - z1_.transpose(-1, -2).matmul(z2_))
+        cx = torch.cos(x) / ell  #(... n x d x mx)
+        cy = torch.cos(y) / ell
+        sx = torch.sin(x) / ell
+        sy = torch.sin(y) / ell
+        z1_ = torch.cat([cx, sx], dim=-2)  #(... n x 2d x mx)
+        z2_ = torch.cat([cy, sy], dim=-2)  #(... n x 2d x mx)
+        const = d * (torch.square(ell**(-1))).mean(
```

Comment:
I think it might be simpler to write `const = ell.square().reciprocal().sum(-2)`

Reply:
yea so as discussed offline, what I wrote is wrong and should probably be 
`const = d * ell.square().reciprocal().mean(-2))`